arbitrary-users-patch/base_images

```diff
@@ -1,6 +1,7 @@
 che-python-3.6          centos/python-36-centos7:1
 che-php-7               eclipse/php:7.1-che7
 che-golang-1.10         golang:1.10.7-stretch
+che-golang-1.12         golang:1.12-stretch
```

Comment:
Since our current golang devfile is just used for the one project, we should ensure `golang:1.12-stretch` works with https://github.com/golang/example.git and replace `che-golang-1.10` with `che-golang-1.12`. I'd rather not build more images than is necessary here.